Zend/zend_fibers.h

```diff
@@ -72,7 +72,10 @@ typedef zend_fiber_context *(*zend_fiber_coroutine)(zend_fiber_context *context)
 
 /* Defined as a macro to allow anonymous embedding. */
 #define ZEND_FIBER_CONTEXT_FIELDS \
+	/* Handle to fiber state as needed by boost.context */ \
 	void *handle; \
+	/* Pointer that identifies the fiber type. */ \
+	void *kind; \
```

Comment:
Should this use `zend_class_entry *`, if a CE is used for identification?

Reply:
This is specified as `void *` to support fibers that are not exposed to PHP userland (pure C fibers) aswell hence restricting it to `zend_class_entry *` would be difficult due to no class entry being present. In essence this pointer just needs to be a unique address for each type of fiber. I decided to use `zend_fiber_ce` because it satisfies this requirement and is easy to associate with the fiber.